pkg/infra/kubernetes/exec_unit.go

```diff
@@ -30,6 +33,52 @@ type HelmExecUnit struct {
 	HorizontalPodAutoscaler *core.SourceFile
 }
 
+type manifestTransformer[K runtime.Object] struct {
+	fieldToTransform func(unit *HelmExecUnit) *core.SourceFile
+	transform        func(unit *HelmExecUnit, cfg config.ExecutionUnit, target K, log *zap.SugaredLogger) ([]HelmChartValue, error)
+	// readF is the function to read a given file to the runtime.Object that we will then cast down to K. You may leave
+	// it blank, in which case the transformer will assume readFile.
+	readF func(f *core.SourceFile) (runtime.Object, error)
+}
+
+func (transformer manifestTransformer[K]) apply(unit *HelmExecUnit, cfg config.ExecutionUnit) ([]HelmChartValue, error) {
+	source := transformer.fieldToTransform(unit)
+	log := zap.L().Sugar().With(logging.FileField(source), zap.String("unit", unit.Name))
+	log.Debugf("Transforming file, %s, for exec unit, %s", source.Path(), unit.Name)
```

Comment:
nit: the unit and source are redundant since they're already fields on the logger.

Reply:
This is what they were before, so I'll keep them for now — I don't feel like investigating what the UX difference would be. :)